lib/browser/api/web-contents.ts

```diff
@@ -553,43 +577,60 @@ WebContents.prototype._init = function () {
   if (this.getType() !== 'remote') {
     // Make new windows requested by links behave like "window.open".
     this.on('-new-window' as any, (event: any, url: string, frameName: string, disposition: string,
-      rawFeatures: string, referrer: string, postData: string) => {
-      const { options, webPreferences, additionalFeatures } = parseFeatures(rawFeatures);
-      const mergedOptions = {
-        show: true,
-        width: 800,
-        height: 600,
-        title: frameName,
-        webPreferences,
-        ...options
-      };
+      rawFeatures: string, referrer: any, postData: any) => {
+      openGuestWindow({
+        event,
+        embedder: event.sender,
+        disposition,
+        referrer,
+        postData,
+        overrideBrowserWindowOptions: {},
+        windowOpenArgs: {
+          url,
+          frameName,
+          features: rawFeatures
+        }
+      });
+    });
 
-      internalWindowOpen(event, url, referrer, frameName, disposition, mergedOptions, additionalFeatures, postData);
+    let windowOpenOverriddenOptions: BrowserWindowConstructorOptions | null = null;
+    this.on('-will-add-new-contents' as any, (event: any, url: string, frameName: string) => {
+      process.stdout.write('test\ntest\nhere\ncool\n');
+      windowOpenOverriddenOptions = this._callWindowOpenOverride(event, url, frameName);
+      if (!event.defaultPrevented && windowOpenOverriddenOptions) {
+        this._setNextChildWebPreferences({
```

Comment:
Same question as above, I'd be way more comfortable with this if we had a test that called `window.open` like 50 times in a row (synchronously) and asserts that the webPrefs end up on the right window.